main/org.thingpedia.covid-vaccine/index.js

```
@@ -0,0 +1,91 @@
+            date = new Date();
+        if (distance == undefined || distance == null)
+            distance = 10.0;
+        console.log(location, date, distance);
+
+        // TODO: Find a better way to mock.
+        if (process.env.CI == true) {
+            return [
+                {
+                  name: 'Safeway 2948',
+                  address: '645 San Antonio Rd',
+                  city: 'Mountain View',
+                  state: 'CA',
+                  postal_code: '94040',
+                  url: 'https://www.safeway.com/pharmacy/covid-19.html'
+                },
+            ]
+        }
+
+        const client = await this._mongodb_client();
+        try {
+            await client.connect();
+            const db = client.db(DB_NAME);
+            const coll = db.collection(COLLECTION_NAME);
+
+            const query = {
+                "geometry.coordinates": {
+                    $near: {
+                        $geometry: {
+                            type: "Point",
+                            coordinates: [location.x, location.y]
+                        },
+                        $maxDistance: distance * 1610,
+                    },
+                },
+                "properties.appointments_available": true
+            };
+
+            const cursor = coll.find(query);
+            let appointments = await cursor.toArray();
+            appointments = appointments.map((appt) => {
+                return {
+                    name: appt.properties.name,
+                    address: appt.properties.address,
+                    city: appt.properties.city,
+                    state: appt.properties.state,
+                    postal_code: appt.properties.postal_code,
+                    url: appt.properties.url
+                };
+            });
+
+            await cursor.close();
+            console.log(appointments);
+            return appointments;
+        } catch (error) {
```

Let the error bubble through, and handle it with `#_[on_error]`, or let the agent reply that there was an unexpected error.

---

Done